tools/check-templates

```diff
@@ -43,11 +44,19 @@ def check_html_templates(templates, modified_only):
     # The casperjs files use HTML5 (whereas Zulip prefers XHTML), and
     # there are also cases where Casper deliberately uses invalid HTML,
     # so we exclude them from our linter.
+    templates_casperjs_filtered = filter(
+        lambda fn: ('casperjs' not in fn),
+        templates) # type: (Iterable[str])
+
     templates = filter(
-        lambda fn: ('base.html' not in fn) and ('casperjs' not in fn),
-        templates)
+        lambda fn: ('base.html' not in fn),
+        templates_casperjs_filtered)
+
```

Comment:
So should we leave out base.html on checking for id's or should we append its path to templates before passing it to html_branches separately if we were to have only one templates list.
Also i wanted to ask the need of running sorted on templates list because from what i understand it shouldn't matter if the list is sorted or not.